rpc/client/options_test.go

```diff
@@ -0,0 +1,158 @@
+package client
+
+import (
+	"crypto/tls"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestWithNetworkURIAddress(t *testing.T) {
+	testCases := []struct {
+		uri string
+
+		wantHost string
+		wantTLS  bool
+	}{
+		{
+			uri:      "grpc://neofs.example.com:8080",
+			wantHost: "neofs.example.com:8080",
+			wantTLS:  false,
+		},
+		{
+			uri:      "grpcs://neofs.example.com:8080",
+			wantHost: "neofs.example.com:8080",
+			wantTLS:  true,
+		},
+		{
+			uri:      "wrongScheme://neofs.example.com:8080",
+			wantHost: "",
+			wantTLS:  false,
+		},
+		{
+			uri:      "impossibleToParseIt",
+			wantHost: "",
+			wantTLS:  false,
+		},
+	}
+
+	for _, test := range testCases {
+		cfg := &cfg{}
+		opt := WithNetworkURIAddress(test.uri)
+
+		opt(cfg)
+
+		require.Equal(t, cfg.addr, test.wantHost, test.uri)
+		require.Equal(t, cfg.tlsCfg != nil, test.wantTLS, test.uri)
```

Comment:
Also double check on argument orders in `require.Equal`. The second argument is `Expected` and third is `Got`. In this test it is in a wrong order.

Reply:
weird mistake, thank you fixed